diff -ur xyssl-0.9-bsd/include/xyssl/config.h xyssl-0.9-bsd+nobloat/include/xyssl/config.h
--- xyssl-0.9-bsd/include/xyssl/config.h	2008-03-16 15:23:22.000000000 +0100
+++ xyssl-0.9-bsd+nobloat/include/xyssl/config.h	2008-04-10 15:14:03.000000000 +0200
@@ -32,8 +32,9 @@
 
 /*
  * Uncomment to enable the use of assembly code.
- */
+ *
 #define XYSSL_HAVE_ASM
+ */
 
 /*
  * Uncomment if the CPU supports SSE2 (IA-32 specific).
@@ -43,13 +44,15 @@
 
 /*
  * Enable all SSL/TLS debugging messages.
- */
+ *
 #define XYSSL_DEBUG_MSG
+ */
 
 /*
  * Enable the checkup functions (*_self_test).
- */
+ *
 #define XYSSL_SELF_TEST
+ */
 
 /*
  * Enable the prime-number generation code.
@@ -107,8 +110,9 @@
  * Caller:
  *
  * This module is used for testing (ssl_client/server).
- */
+ *
 #define XYSSL_CERTS_C
+ */
 
 /*
  * Module:  library/debug.c
@@ -117,8 +121,9 @@
  *          library/ssl_tls.c
  *
  * This module provides debugging functions.
- */
+ *
 #define XYSSL_DEBUG_C
+ */
 
 /*
  * Module:  library/des.c
@@ -127,8 +132,9 @@
  * This module enables the following ciphersuites:
  *      SSL_RSA_DES_168_SHA
  *      SSL_EDH_RSA_DES_168_SHA
- */
+ *
 #define XYSSL_DES_C
+ */
 
 /*
  * Module:  library/dhm.c
@@ -138,8 +144,9 @@
  * This module enables the following ciphersuites:
  *      SSL_EDH_RSA_DES_168_SHA
  *      SSL_EDH_RSA_AES_256_SHA
- */
+ *
 #define XYSSL_DHM_C
+ */
 
 /*
  * Module:  library/havege.c
@@ -189,8 +196,9 @@
  * Caller:  library/aes.c
  *
  * This modules adds support for the VIA PadLock on x86.
- */
+ *
 #define XYSSL_PADLOCK_C
+ */
 
 /*
  * Module:  library/rsa.c
@@ -219,16 +227,18 @@
  * Caller:
  *
  * This module adds support for SHA-224 and SHA-256.
- */
+ *
 #define XYSSL_SHA2_C
+ */
 
 /*
  * Module:  library/sha4.c
  * Caller:
  *
  * This module adds support for SHA-384 and SHA-512.
- */
+ *
 #define XYSSL_SHA4_C
+ */
 
 /*
  * Module:  library/ssl_cli.c
Only in xyssl-0.9-bsd+nobloat/include/xyssl: config.h~
Only in xyssl-0.9-bsd+nobloat/library: Makefile~
